airflow/models/dagbag.py

```diff
@@ -391,6 +396,14 @@ def collect_dags(
             'dagbag_import_errors', len(self.import_errors), 1)
         self.dagbag_stats = sorted(
             stats, key=lambda x: x.duration, reverse=True)
+        for file_stat in self.dagbag_stats:
+            dag_ids = dags_by_name[file_stat.dags]
+            if file_stat.dag_num >= 1:
+                # if we found multiple dags per file, the stat is 'dag_id1 _ dag_id2'
+                dag_names = '_'.join(dag_ids)
```

Comment:
hey @XD-DENG / @feng-tao / @milton0825 
about the `dag_names = '_'.join(dag_ids)`; we have some files generating 100s of dags, which makes this metric quite unusable.
- could you advise how does statsd handles such a long metric name (we are actually using a [statsd to prometheus exporter](https://github.com/prometheus/statsd_exporter) so im not that familiar with statsd)?
- is it not possible to get the name of the file where the dags are generated instead?
- the description in docs/metrics.rst doesn't quite match the current implementation

Reply:
I think it may make more sense to name the metrics by the filename given that there can be a lot of dynamic DAGs that come out of a single file.

Reply:
Can you file a ticket in https://issues.apache.org/jira/projects/AIRFLOW/issues?

Reply:
cheers - https://issues.apache.org/jira/browse/AIRFLOW-5274

Reply:
@marengaz make sense. I could take this one.

Reply:
@milton0825 @marengaz https://github.com/apache/airflow/pull/5890